vertical-pod-autoscaler/enhancements/7571-support-pod-level-resources/README.md

```diff
@@ -0,0 +1,253 @@
+- Sidecars, such as logging agents or mesh proxies (like `tool1` or `tool2`), that don't use container-level limits can borrow idle CPU from other containers in the pod when they experience a spike in usage. Pod-level resources allow a container experiencing a spike to access idle resources from others, optimizing overall utilization.
+
+**Cons**:
+- Applying both pod-level and container-level recommendations requires eviction because [in-place pod-level resource resizing](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/5419-pod-level-resources-in-place-resize) is not yet available.
+- This option adds complexity: VPA must track which container-level resources are under its control by default and avoid mutating others.
+- Existing VPA users may find the behavior surprising because VPA does not control all container-level resources stanzas - only those initially set - unless configured otherwise.
+
+#### No resources stanza exists at the pod and container level
+
+When a workload is created without any resources defined at either the pod or container level, there are two options:
+
+##### Option 1: VPA controls only the container-level resources
+
+This option mirrors current VPA behavior by managing only container-level resources, preserving benefits like in-place container-level resource resize. In this mode, pod-level recommendations are not computed and therefore not applied.
+
+**Pros**:
+- Familiar to existing users because it does not change current VPA behavior.
+
+**Cons**:
+- No cross-container resource sharing: in multi-container Pods, a container can hit its own limit and be throttled even if sibling containers are idle.
+
+##### Option 2: VPA controls only the pod-level resources
+
+With this option, VPA computes and applies only pod-level recommendations.
+
+**Pros**:
+- Enables shared headroom across containers in the same Pod, previously with container-only limits, a sidecar (e.g. `tool1` or `tool2` container) hitting its own CPU limit could throttle the Pod even if other containers had spare CPU. Pod-level resources allows one container experiencing a spike to access idle resources from others, optimizing overall utilization.
+- Simple to adopt and remains straightforward for users if documented clearly in official documentation.
+
+**Cons**:
+- [In-place pod-level resource resize](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/5419-pod-level-resources-in-place-resize) is not yet available, so the updater would need to evict Pods for the admission controller to apply pod-level recommendations, future proposals can add in-place support when available.
+
+## Proposal
+
+- Add a new feature flag named `PodLevelResources`. Because this proposal introduces new code paths across all three VPA components, this flag will be added to each component.
```

Comment:
Is this a feature flag to assist with GAing the feature, or is it a flag to enable/disable the feature?